packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/entity_type/mod.rs

```diff
@@ -22,10 +23,11 @@ impl<C: AsClient> EntityTypeStore for PostgresStore<C> {
                 .change_context(InsertionError)?,
         );
 
-        let (version_id, identifier) = transaction.create(entity_type, created_by).await?;
+        // TODO - get rid of the clone on entity_type
```

Comment:
We extract the references from the type afterwards. I guess we could fix this by changing `insert_entity_type_references` to take the extracted contents, and extract them before creating the entity_type in the DB.

That will use up more memory, but I'm not sure it's a problem.

Reply:
I think we can optimize it by creating a transaction, disabling the integrity checks for the transaction's lifetime, inserting the reference, and then creating the type. However, I think the bottleneck here will be the PG connection, not the cloning. I'd leave it as it is now with the TODO in place. I guess we will change this function at some point anyway, so we should avoid premature optimization.

Reply:
I have removed the TODO as I don't think we should put TODOs for optimizations that aren't backed by evidence. I have replaced it with a comment explaining the clone isn't really needed though.